src/rust/engine/sharded_lmdb/src/lib.rs

```diff
@@ -166,15 +172,20 @@ impl ShardedLmdb {
     Ok(ShardedLmdb {
       lmdbs,
       root_path,
-      max_size,
+      num_shards,
+      max_size_per_shard,
       executor,
       lease_time,
     })
   }
 
-  fn envs(root_path: &Path, max_size: usize) -> Result<Vec<(Environment, PathBuf, u8)>, String> {
-    let mut envs = Vec::with_capacity(0x10);
-    for b in 0x00..0x10 {
+  fn envs(
+    root_path: &Path,
+    num_shards: u8,
```

Comment:
The `b << 4` below is a hard-coded assumption that `num_shards` is always 16. I believe this code would be broken if called with `num_shards` not equal to 16?

Given we only ever use one value anyway, maybe just pull out an associated constant `ShardedLmdb::NUM_SHARDS = 0x10`?

Reply:
Yea, good call.